lib/runtime/src/component/endpoint.rs

```diff
@@ -178,37 +183,79 @@ impl EndpointConfigBuilder {
         let component_name_for_task = component_name.clone();
         let endpoint_name_for_task = endpoint_name.clone();
 
-        let task = tokio::spawn(async move {
-            let result = push_endpoint
-                .start(
-                    service_endpoint,
-                    namespace_name_for_task,
-                    component_name_for_task,
-                    endpoint_name_for_task,
-                    connection_id,
-                    system_health,
-                )
-                .await;
+        // Get the unified request plane server (works for all transport types)
+        let server = endpoint.drt().request_plane_server().await?;
+
+        tracing::info!(
+            endpoint = %endpoint_name_for_task,
+            transport = server.transport_name(),
+            "Registering endpoint with request plane server"
+        );
+
+        // Register endpoint with the server (unified interface)
+        server
+            .register_endpoint(
+                endpoint_name_for_task.clone(),
+                handler,
+                connection_id,
+                namespace_name_for_task.clone(),
+                component_name_for_task.clone(),
+                system_health.clone(),
+            )
+            .await?;
```

Comment:
ok. we are registering to a shared instance.  i had a comment that was pending regarding what looked to be a http server to endpoint, but i couldn't quite resolve how it was being called.  if that still exist in the review we can resolve it.